### DISCUSSION TITLE
Problem when having a checkbox in a form and using a pydantic model.

### DISCUSSION THREAD

**Discussion by arminvburren** [fastapi/fastapi#9409] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
https://stackoverflow.com/questions/73078330/fastapi-request-body-returns-null-from-checkbox-html-tag
```

### Description

Hello, I'm having exactly the problem discussed here: https://stackoverflow.com/questions/73078330/fastapi-request-body-returns-null-from-checkbox-html-tag 

And the solution given on StackOverflow just doesn't work.

If you have a form and you declare each parameter in the function such as the following it works. 

```
@app.post("/test")
async def my_post(request: Request,
                          date: str = Form(...),
                          mychecked: bool = Form(False), 
                          ):
```

But if you want to use a pydantic model 

```
@app.post("/test")
async def my_post(request: Request,
                           my_inputs: MyInputs = Depends(MyInputs),
                          ):
```
and a model like that :
```
@form_body
class MyInputs(BaseModel):
    date          = ""
    mychecked   = "False"

```
Then it just doesn't work when the checkbox is unchecked. There's always "value error missing".

I've tried serveral solutions including the SO one, and two other given by ChatGPT inculding hacking the validator of the pydantic model. But none of it works :(.

### Operating System

Linux

### FastAPI Version

0.79.0

### Python Version

3.8

**Comment by chrisK824** [comment]
Can you provide all the piece of code you are using? I suppose you are using a decorator from one of the answers in stackoverfow questions?

Btw there seems to be some wrong syntax in you model, should be something like this (assuming the correct data types)

```
@form_body
class MyInputs(BaseModel):
    date:  Optional[str] = ""
    mychecked: Optional[bool]  = False
```
